### DISCUSSION TITLE
Add reviewers/participants as co-authors of PR

### DISCUSSION THREAD

**Discussion by tupui** [community/community#4525] [Pull Requests]
Reviewing a PR is arguably a lot of work. If you do not do any suggestion using the `Insert a suggestion` feature, you will not be listed in the co-authors and not appear in the git history.

It would be great to automatically add co-authoring when the PR is getting merged (for instance). As when doing a squash merge, you could always remove this pre-filled information (or could be a config in the project settings) if you do not like this.

In general, I would even push this with adding people from the discussion. In some cases, you have invaluable information here.

I am not sure if we could, but instead of using `Co-authored-by`, it could write something else like `Reviewed-by`. This could help automated tools.

```
This was a great collaborative work

Co-authored-by: bob <...>
Reviewed-by: bob <...>
Discussed-with: bob <...>
```

Alternatively, you could just add a bit more information in commit message like for instance
```
This was a great collaborative work

Co-authors:
Co-authored-by: bob <...>
Reviewers:
Co-authored-by: bob <...>
Discussions:
Co-authored-by: bob <...>
```

I think this kind of information can be helpful and it adds valuable information in the git history. It is clear then who validated a new features, etc. Last but not least, it helps with giving credits to people. Helping on the issue tracker is a real contribution to a project and should be reflected in the Git history too.

**Comment by jjerphan** [comment]
Thank you for making this proposal, @tupui. I think coming up with a way to configure co-authors addition in merge commit would really be appreciable. I also think that triage work for issues (understanding/identifying the problem reported) aren't valued enough and that triagers should be traced in the history.

What if trailers [as used for the Linux Kernel and other projects' development](https://git.wiki.kernel.org/index.php/CommitMessageConventions) could be supported? 

Building on top of @tupui last example, this would give the following commit message appendice:

```
This was a great collaborative work

Reported-by: Alice <...>
Reviewed-by: Bob <...>
Discussed-by: Charly <...>
Co-authored-by: Dan <...>
```

Moreover and in term of usability, could a template for merge commit message appendices be included in the project history and be tracked -- similarly to the `CODEOWNERS` file -- for the configuration of such a feature?
